sdl-core/src/test/scala/io/smartdatalake/workflow/dataobject/SFtpFileRefDataObjectTest.scala

```diff
@@ -207,4 +210,110 @@ class SFtpFileRefDataObjectTest extends FunSuite with Matchers with BeforeAndAft
     val fileRefs2 = sftpDO.getFileRefs(Seq())
     assert(fileRefs2.size == 2 && fileRefs2.map(_.fileName).forall(_.startsWith(resourceFile)))
   }
+
+
+  test("overwrite target") {
+
+    val srcDir1 = "testSrc1"
+    val srcDir2 = "testSrc2"
+    val tgtDir = "testTgt"
+    val resourceFile = "AB_NYC_2019.csv"
+
+    // copy data file to ftp
+    TestUtil.copyResourceToFile(resourceFile, tempDir.resolve(srcDir1).resolve(resourceFile + "1").toFile)
+    TestUtil.copyResourceToFile(resourceFile, tempDir.resolve(srcDir2).resolve(resourceFile + "2").toFile)
+
+    // setup DataObject
+    val srcDO1 = SFtpFileRefDataObject("src1", tempDir.resolve(srcDir1).toString.replace('\\', '/'), connectionId = "con1")
+    val srcDO2 = SFtpFileRefDataObject("src1", tempDir.resolve(srcDir2).toString.replace('\\', '/'), connectionId = "con1")
+    val tgtDO = SFtpFileRefDataObject("tgt1", tempDir.resolve(tgtDir).toString.replace('\\', '/'), connectionId = "con1")
+
+    // src1 file transfer
+    val filetransfer1 = new StreamFileTransfer(srcDO1, tgtDO)
+    val srcFileRefs1 = srcDO1.getFileRefs(Seq())
+    assert(srcFileRefs1.nonEmpty)
+    val fileRefPairs1 = tgtDO.translateFileRefs(srcFileRefs1)
+    tgtDO.startWritingOutputStreams(fileRefPairs1.map(_.tgt.partitionValues))
+    filetransfer1.exec(fileRefPairs1)
+    val tgtFileRefs1 = tgtDO.getFileRefs(Seq())
```

Comment:
is this calling translateFileRefs under the hood?
If so, you are just testing that translateFileRefs is called.
It would be a bit easier to follow with concrete Filename Strings to understand what s happenning

Reply:
You're right. And making some test conditions more explicit, this could also help testing more functionality.
Added a commit for that.